src/email-verification/email-verification.service.ts

```diff
@@ -80,9 +84,10 @@ export class EmailVerificationService {
    * @Params: token - token sent to the user
    * return - return a called function(createUserProfile) to create user profile
    */
-  async decodeEmailToken(token: string) {
+  async decodeEmailToken(userId: string) {
     try {
-      const payload = await this.jwtService.verify(token, {
+      const {signupToken} = await this.usersRepo.findOne({where:{id: userId}})
+      const payload = await this.jwtService.verify(signupToken, {
         secret: this.configService.get(configConstant.jwt.verify_secret),
       });
       // Mark User email as verified
```

Comment:
Don't you need to check the outcome of this verification before proceeding?

Reply:
yes it is being checked... it is being handled in the catch block